src/variant.rs

```diff
@@ -165,11 +169,11 @@ impl Variant {
             VT_EMPTY => Variant::Empty,
             VT_NULL => Variant::Null,
             VT_UNKNOWN => {
-                let unk = unsafe { &vt.Anonymous.Anonymous.Anonymous.punkVal };
-                let ptr = unk.as_ref().ok_or(WMIError::NullPointerResult)?;
+                let unk = unsafe { &(vt.Anonymous.Anonymous.Anonymous.punkVal as *const IUnknown) };
+                let ptr = unsafe { unk.as_ref() }.ok_or(WMIError::NullPointerResult)?;
```

Comment:
I think both lines can be simplified to 
`let ptr = unsafe { IUnknown::from_raw(vt.Anonymous.Anonymous.Anonymous.punkVal) };`